primitiv/msgpack.h

```diff
@@ -205,12 +205,7 @@ class Writer : mixins::Nonmovable<Writer> {
     return *this;
   }
 
-  Writer &operator<<(const char *x) {
-    return operator<<(std::string(x));
-  }
-
-  Writer &operator<<(const std::string &x) {
-    const std::uint64_t size = x.size();
+  Writer &writeString(const char *x, std::size_t size) {     
```

Comment:
* `writeString` -> `write_string` (we are using snake_case for most functions).
* Better to make this function private to hide the interface to directly store binaries to the `str` message.